alphagov/paas-bootstrap
# [#161591823] Use bosh-deployment for bosh manifest

What
----

We want to migrate paas-bootstrap bosh-manifest to use bosh-deployment in a similar way we did with cf-deployment in paas-cf.

https://github.com/cloudfoundry/bosh-deployment

In this PR we implement a model similar to the paas-cf prometheus manifest, with a submodule for bosh-deployment.

We work based on https://github.com/alphagov/paas-bootstrap/pull/231 to achieve a nop manifest. 

We move the secret and cert generation into `bosh interpolate`, so that the secrets and certs would be stored in the `bosh-vars-store.yml`. 

We adapt the rotation of secrets, but do not fix the rotation of the bosh-CA. Bosh CA would still be generated using certstrap.

In this PR we do not want to upgrade any software, so we pin the versions to older versions, only for the sake of migrating the pipeline.

See commits for details.

Dependencies
-------------------

Review with https://github.com/alphagov/paas-bootstrap/pull/231 

If https://github.com/alphagov/paas-bootstrap/pull/231 is merged, then rebase this PR.

After: https://github.com/alphagov/paas-bootstrap/pull/232 and https://github.com/alphagov/paas-cf/pull/1695


How to review
-------------

- Review with https://github.com/alphagov/paas-bootstrap/pull/231 
- Code review.
- Check the script `./diff-manifest.sh init` and `./diff-manifest.sh diff` to check that  the manifest is equivalent. The file itself would be different, so the bosh-deploy would redeploy the VM regardless.
- Check the secret rotation: For that you need to revert the migrate commit!
- Remove any WIP commit.

This script https://gist.github.com/keymon/6e38941995a48af109e28192bff6c9cb might be handy to compare versions of files in S3. 
 
Who can review
--------------

Not me.
